Deprecate using `GET /api/v1/fleet/commands` w/o host_identifier (#44392)
fleetdm/fleet: docs/REST API/rest-api.md

| per_page                  | integer | query | Results per page. Default is `10`.                                        |
| order_key                 | string  | query | What to order results by. Can be any field listed in the `results` array example below. Default is `updated_at`. |
| order_direction           | string  | query | **Requires `order_key`**. The direction of the order given the order key. Options include `"asc"` and `"desc"`. Default is `"asc"`. |

> | host_identifier           | string  | query | **Required for new integrations.** The host's `hostname`, `uuid`, or `hardware_serial`. Returns only commands that target the specified host. Omitting `host_identifier` is deprecated (see the deprecation notice below) and will be rejected in a future Fleet release. |

Reviewer: What does "required for new integrations" mean? Does it error now if not included, or is this just saying it will error in the next major release?

Author: Next major release.

Reviewer 2: I think we can just saw **Required** and mention that it will be removed in Fleet 5. I also moved all the important info to the top (along w/ the required key): https://github.com/fleetdm/fleet/pull/44392/changes#diff-7246bc304b15c8865ed8eaa205e9c244d0a0314e4bae60cf553dc06147c38b64R7688

What do y'all think?

> Apple (macOS, iOS, iPadOS) MDM commands that 'ran' have an 'Acknowledged' `status`. Commands that are 'pending' have a 'Pending' or 'NotNow' `status`. Apple commands that 'failed' have an 'Error' `status`.
>
> Apple (macOS, iOS, iPadOS) InstallProfile and RemoveProfile commands enqueued by Fleet going forward will have a non-`null` "name" which represents the profile name. Previously-enqueued(prior to v4.84.0) or manually-enqueued commands will have a `null` name, as will other types of Apple MDM commands and all Windows commands.
>

> > **Deprecated:** calling this endpoint without `host_identifier` is deprecated and will be required in a future Fleet release. Always provide `host_identifier` to scope the response to a single host. Unscoped calls continue to work for backward compatibility but are not recommended; the underlying query is not performant at scale and may time out on large fleets. New integrations must pass `host_identifier`.

Reviewer 3: ```suggestion
> **Deprecated:** This endpoint will be replaced in the next major release. Calling this endpoint without ⁠`host_identifier` is **not recommended**. The underlying query is not performant at scale and may time out.
```
